sdk/core/azure-core/src/test/java/com/azure/core/UserAgentTests.java

```diff
@@ -57,4 +58,61 @@ public void customUserAgentTests() throws Exception {
                 new URL("http://localhost"))).block();
         Assertions.assertEquals(200, response.getStatusCode());
     }
+
+    @Test
+    public void defaultApplicationIdUserAgentTest() throws Exception {
+        final HttpPipeline pipeline = new HttpPipelineBuilder()
+            .httpClient(new NoOpHttpClient() {
+                @Override
+                public Mono<HttpResponse> send(HttpRequest request) {
+                    String header = request.getHeaders().getValue("User-Agent");
+                    String expectedHeaderPrefix = "azsdk-java-package.name";
+                    Assertions.assertTrue(header.startsWith(expectedHeaderPrefix));
+                    return Mono.just(new MockHttpResponse(request, 200));
+                }
+            })
+            .policies(new UserAgentPolicy(
+                null,
+                "package.name",
+                "package_version",
+                Configuration.NONE,
+                () -> "1.0"))
+            .build();
+        HttpResponse response = pipeline.send(new HttpRequest(HttpMethod.GET,
+            new URL("http://localhost"))).block();
+        Assertions.assertEquals(200, response.getStatusCode());
```

Comment:
I would also assert that the `response` contains the headers you expect to exist?